crucible-jvm/src/Lang/JVM/JavaTools.hs

```diff
@@ -75,12 +78,12 @@ findJavaProperty javaPath propertyNeedle = do
 
 -- | @'findJavaMajorVersion' javaPath@ will consult @javaPath@ to find the
 -- major version number corresponding to that Java release.
-findJavaMajorVersion :: FilePath -> IO Int
+findJavaMajorVersion :: FilePath -> IO (Maybe Int)
```

Comment:
Rather than returning `Maybe Int`, I think we should return something akin to `Either Text Int`. That way, `crucible-jvm` can report what went wrong in the event that `findJavaMajorVersion` fails.

Reply:
We don't really know what went wrong beyond that we couldn't get the version (i.e. the error is always "parse error").  I guess I could return the output of the command, and we could show that to the user, although I wonder how useful it is unless we show them how run the command as well...

Reply:
Ah, sorry, to be clear: I think we should also change the return type of `findJavaProperty` to return `Either Text` so that we can display the output of the command.